platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -161,53 +169,158 @@ public BaseEventHashedData(
                 .toArray(new ConsensusTransactionImpl[0]);
     }
 
-    @Override
-    public int getMinimumSupportedVersion() {
-        return ClassVersion.BIRTH_ROUND;
+    /**
+     * Serialize the event for the purpose of creating a hash.
+     * Since events are being migrated to protobuf, calculating the hash of an event will change.
+     *
+     * @param out the stream to which this object is to be written
+     * @throws IOException if unsupported payload types are encountered
+     */
+    public void serializeLegacyHashBytes(final SerializableDataOutputStream out) throws IOException {
+        out.writeLong(CLASS_ID);
+        serialize(out);
     }
 
-    @Override
+    /**
+     * Serialize the event
+     *
+     * @param out the stream to which this object is to be written
+     *
+     * @throws IOException if unsupported payload types are encountered
+     */
     public void serialize(final SerializableDataOutputStream out) throws IOException {
+        out.writeInt(ClassVersion.BIRTH_ROUND);
         out.writeSerializable(softwareVersion, true);
-        out.writeSerializable(creatorId, false);
+        out.writeInt(NodeId.ClassVersion.ORIGINAL);
+        out.writeLong(eventCore.creatorNodeId());
         out.writeSerializable(selfParent, false);
         out.writeSerializableList(otherParents, false, true);
-        out.writeLong(birthRound);
-        out.writeInstant(timeCreated);
+        out.writeLong(eventCore.birthRound());
+        out.writeInstant(HapiUtils.asInstant(eventCore.timeCreated()));
 
         // write serialized length of transaction array first, so during the deserialization proces
         // it is possible to skip transaction array and move on to the next object
-        out.writeInt(getSerializedLength(transactions, true, false));
+        out.writeInt(PayloadUtils.getLegacyObjectSize(payloads));
         // transactions may include both system transactions and application transactions
         // so writeClassId set to true and allSameClass set to false
-        out.writeSerializableArray(transactions, true, false);
+        out.writeInt(payloads.size());
+        if (!payloads.isEmpty()) {
+            final boolean allSameClass = false;
```

Comment:
What is the point of `allSameClass`, if it's only ever `false`, or absent?

Reply:
Wanted to make clear what this `false` is for